### PR TITLE
Non-repudiation: documentation

### LINKED ISSUE #8659: Non-repudiation: documentation
This should be the final ticket to address in this milestone: once ready for early access, the non-repudiation middleware and clients should be properly documented.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
[Runtime Components] New feature: the non-repudiation middleware is now available
as an early access feature, please have a look at its page in the early access
section of the documentation to learn more
changelog_end

Closes #8659

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/tools/non-repudiation.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Non-repudiation
+###############
```
Comment (cocreature) [inline docs/source/tools/non-repudiation.rst:5]: We should probably copy the early access & EE disclaimer from the profiler here as well. I know you mention it below but it seems nice to be consistent and have this in the same place everywhere.

File: docs/source/tools/non-repudiation.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Non-repudiation
```
Comment (cocreature) [inline docs/source/tools/non-repudiation.rst:4]:
```suggestion
Non-Repudiation
```
the DAML

File: docs/source/tools/non-repudiation.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+The non-repudiation system consists of three components:
+
+- the non-repudiation middleware is a reverse proxy that sits in front of the Ledger API that verifies command signatures and forwards the signed command to the actual participant node
+- the non-repudiation API is a web server used by the operator to upload new certificates and verify repudiation claims
+- the non-repudiation client is a gRPC interceptor that can be used alongside any gRPC client on the JVM, including the official Java bindings, that will ensure that commands are signed with a given private key
+
+Running the server-side components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The server-side components are the middleware and the API. Both can be run as a single process by running the non-repudiation fat JAR provided as part of Daml Connect Enterprise Edition.
```
Comment (cocreature) [inline docs/source/tools/non-repudiation.rst:31]:
```suggestion
The server-side components are the middleware and the API. Both are run as a single process by running the non-repudiation fat JAR provided as part of Daml Connect Enterprise Edition.
```
You don’t really have an option to run them separately atm aafaik.

File: docs/source/tools/non-repudiation.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Note that at the current stage you need to also have a PostgreSQL server running where signed commands will be persisted.
```
Comment (cocreature) [inline docs/source/tools/non-repudiation.rst:33]: The lawn mower does not approve

File: docs/source/tools/non-repudiation.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    builder.intercept(SigningInterceptor.signCommands(key, certificate));
+    DamlLedgerClient client = DamlLedgerClient.newBuilder(builder).build();
+    client.connect();
+
+Non-repudiation over the HTTP JSON API
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note that the current status is that non-repudiation applies exclusively to Ledger API messages: if you want to use the HTTP JSON API you will need to
+run your own HTTP JSON API server and start it with a certificate that will be used to sign every command issued by the HTTP JSON API to the participant.
+This means that running the HTTP JSON API with non-repudiation as a ledger operator is currently not supported.
```
Comment (cocreature) [inline docs/source/tools/non-repudiation.rst:66]: I’d rephrase this slightly. It is supported and you can come up with cases where this is sensible, e.g., if you don’t expose it publically. The crucial point is that you don’t get non-repudiation for the leg user <-> http json API.

**Review by cocreature — APPROVED** [review]
Nice work, thanks!

File: docs/source/tools/non-repudiation.rst (hunk 2 above)
Comment (stefanobaghino-da) [reply docs/source/tools/non-repudiation.rst:4]: Are you sure? I don't think you are supposed to capitalize hyphenated names (see https://en.wikipedia.org/wiki/Non-repudiation).

File: docs/source/tools/non-repudiation.rst (hunk 4 above)
Comment (stefanobaghino-da) [reply docs/source/tools/non-repudiation.rst:33]: Took me a few seconds to get this. 😂 ☕

File: docs/source/tools/non-repudiation.rst (hunk 2 above)
Comment (cocreature) [reply docs/source/tools/non-repudiation.rst:4]: oh you’re right, nvm

File: docs/source/tools/non-repudiation.rst (hunk 5 above)
Comment (stefanobaghino-da) [reply docs/source/tools/non-repudiation.rst]: Addressed by https://github.com/digital-asset/daml/pull/9288/commits/686fe9c37e8ec0939e35bf477aa1c7c1a7d3af8c

File: docs/source/tools/non-repudiation.rst (hunk 1 above)
Comment (stefanobaghino-da) [reply docs/source/tools/non-repudiation.rst:5]: Addressed by https://github.com/digital-asset/daml/pull/9288/commits/2b27aa7ed0ab4cbbe2db4e2d156811532553c6a5

File: docs/source/tools/non-repudiation.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Non-repudiation
+###############
+
+The Daml Profiler is only available in the
```
Comment (stefanobaghino-da) [inline docs/source/tools/non-repudiation.rst:7]: 🍝 On it.

Comment (stefanobaghino-da) [reply docs/source/tools/non-repudiation.rst]: Addressed by https://github.com/digital-asset/daml/pull/9288/commits/a7458331298a27b9c78f23bf4ba53692a83f1ae9

**Pull request merged**